certora/specs/StkBNB.spec

```diff
@@ -0,0 +1,326 @@
+
+    assert from != to =>
+        balanceOf(e, from) == fromBalanceBefore - amount &&
+        balanceOf(e, to) == toBalanceBefore + amount &&
+        allowanceBefore >= amount &&
+        fromBalanceBefore >= amount &&
+        from != 0 &&
+        to != 0 &&
+        allowance(e, from, e.msg.sender) == allowanceBefore - amount;
+}
+
+rule IsMintPrivileged(address privileged, address recipient, uint256 amount) {
+    env e1;
+	require e1.msg.sender == privileged;
+    bytes userData;
+    bytes operatorData;
+	storage initialStorage = lastStorage;
+    uint256 totalSupplyBefore = totalSupply(e1);
+	mint(e1, recipient, amount, userData, operatorData); // no revert
+	uint256 totalSupplyAfter1 = totalSupply(e1);
+
+    require(totalSupplyAfter1 > totalSupplyBefore);
+
+	env e2;
+	require e2.msg.sender != privileged;
+
+	mint@withrevert(e2, recipient, amount, userData, operatorData) at initialStorage;
+	bool secondSucceeded = !lastReverted;
+    uint256 totalSupplyAfter2 = totalSupply(e2);
+
+	assert  !secondSucceeded || (totalSupplyBefore == totalSupplyAfter2);
+}
+
+rule NoFeeOnTransfer(address bob, uint256 amount) {
+    env e;
+    calldataarg args;
+    require bob != e.msg.sender;
+
+    uint256 balanceSenderBefore = balanceOf(e, e.msg.sender);
+    uint256 balanceBefore = balanceOf(e, bob);
+    transfer(e, bob, amount);
+
+    uint256 balanceAfter = balanceOf(e, bob);
+    uint256 balanceSenderAfter = balanceOf(e, e.msg.sender);
+    assert balanceAfter == balanceBefore + amount;
+}
+
+rule NoFeeOnTransferFrom(address alice, address bob, uint256 amount) {
+    env e;
+    calldataarg args;
+    require alice != bob;
+    require allowance(e, alice, e.msg.sender) >= amount;
+
+    uint256 balanceBefore = balanceOf(e, bob);
+    transferFrom(e, alice, bob, amount);
+
+    uint256 balanceAfter = balanceOf(e, bob);
+    assert balanceAfter == balanceBefore + amount;
+}
+
+rule RevertOnPause(){
+    env e;
+    require paused(e) == true;
+    storage initialStorage = lastStorage;
+    address recipient;
+    uint256 amount;
+    bytes bytesVal;
+    bytes userData;
+    bytes operatorData;
+    address from;
+    address to;
+
+    mint@withrevert(e, recipient, amount, userData, operatorData);
+    assert lastReverted;
+    pause@withrevert(e) at initialStorage;
+    assert lastReverted;
+    burn@withrevert(e, amount, bytesVal) at initialStorage;
+    assert lastReverted;
+    operatorBurn@withrevert(e, from ,amount, bytesVal, bytesVal) at initialStorage;
+    assert lastReverted;
+ }
+
+ rule SendSumOfFromAndToBalancesStaySame(address to, uint256 amount, bytes data) {
+    env e;
+    mathint sum = stakedBNBContract.balanceOf(e,e.msg.sender) + stakedBNBContract.balanceOf(e,to);
+    require sum < max_uint256;
+
+    uint256 totalSupplyBefore = totalSupply(e);
+    send(e, to, amount, data); 
+    uint256 totalSupplyAfter = totalSupply(e);
+
+    assert stakedBNBContract.balanceOf(e,e.msg.sender) + balanceOf(e,to) == sum;
```

Comment:
Did you mean here:
`stakedBNBContract.balanceOf(e,to)` as in line 243
instead of:
`balanceOf(e,to)`?

Reply:
Done